### DISCUSSION TITLE
How to get same $PATH?

### DISCUSSION THREAD

**Discussion by 9mm** [neovide/neovide#2870] [Q&A]
Hey, I think I solved this so long ago but I can't remember at all what I did...

Normally I launch Neovide from ZSH/iTerm3 like this:

```
cd ~/Code/project
n
```

Which launches the `n` bash script:

```bash
#!/bin/bash

export NEOVIDE_FORK=1
export NEOVIDE_FRAME=buttonless
export NEOVIDE_TITLE_HIDDEN=1
export NEOVIDE_VSYNC=0

neovide_bin="/Users/zesty/.cargo/bin/neovide" # "/opt/homebrew/bin/neovide"

current_dir=$(pwd)

if [ -z "$1" ]; then
  if [ "$current_dir" = "$HOME" ]; then
    $neovide_bin -- -c ":cd ~/Desktop"
  else
    $neovide_bin -- -c ":cd $current_dir"
  fi
else
  if [ -d "$1" ]; then
    $neovide_bin -- -c ":cd $1"
  else
    $neovide_bin "$@"
  fi
fi
```

But I also drag Neovide to my dock on rare occassions I just want to open it to take notes in.

I notice when I do `!:echo $PATH` there, it doesn't contain any of my path from `.zshrc`, so a lot of things break.

What is the idiomatic way to fix this?

I am on OSX, and I know there are a lot of `.profile`, `.zshenv` etc. I have tried modifying my path in a few of them and I don't think it did anything, but I wanted to stop shooting in the dark

**Comment by 9mm** [comment]
Hey @falcucci sorry to just @ you out of the blue. if you are busy 100% no pressure to reply to this _at all_, but if you just know off the top of head that would be cool

**Reply by 9mm** [reply]
or at least enough to point me in the right direction of what to search for

I thought about making an automator app but the problem is i can no longer drop files in then. so it actually has to be a real app i think, not some wrapper "launcher"

**Reply by falcucci** [reply]
hey @9mm sorry the delay

could you try to test if it persist using [this](https://github.com/neovide/neovide/actions/runs/12611813099?pr=2928) artifact and try that again?

**Reply by 9mm** [reply]
@falcucci awesome! I will try this tomorrow. Thanks!

**Comment by xav-ie** [comment]
You can change the current PATH in neovim like this:
```
:let $PATH = '/usr/bin:' . $PATH
```

In your current shell do:
```
echo $PATH
```

then in neovim do:
```
q:
```

and do this:
```
let $PATH = <paste what your native shell gave you>
```

^ your path now matches. This is tedious, but it works. I don't know more beyond this.

I think this is general neovim question and not neovide. I would try reddit.

**Reply by 9mm** [reply]
I can change the path inside neovim, the problem is (and why I put it in neovide) is i need it to be on the "app" version thats in my dock. if i make a launcher script with applescript then it wont allow dragging files into the icon). Im trying to make the actual version on my dock default to a specific folder when opened

**Reply by xav-ie** [reply]
I see... you need the path set in startup. I think you can add the let path stuff to your initrc? Or you can add:
```
PATH="/usr/bin:$PATH"
export PATH
```
To the top of your wrapper script

^ I know this method should work, because that is how I do it.
